FakeRate/app_region/ScanChain.C

```diff
@@ -512,147 +523,170 @@ int ScanChain( TChain* chain, TString fakeratefile, TString option = "", TString
-		if (abs(ss.lep1_id())==11) {
-		  pTrelvsIso_histo_el->Fill( std::min(ss.lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_el->Fill( std::min(ss.lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+	    if( ss::lep1_id()*ss::lep2_id() > 0 ) {
+	      if (ss::lep1_motherID()<=0 && /*ss::lep1_iso()>0.1 &&*/ fabs(ss::lep1_ip3d()/ss::lep1_ip3d_err())<4. && ss::lep2_motherID()==1) {
+		if (abs(ss::lep1_id())==11) {
+		  pTrelvsIso_histo_el->Fill( std::min(ss::lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_el->Fill( std::min(ss::lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_el")]->Fill(std::min(lep1_ptrel_v1,float(29.9)) );
 		} else {
-		  pTrelvsIso_histo_mu->Fill( std::min(ss.lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss.lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsIso_histo_mu->Fill( std::min(ss::lep1_iso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss::lep1_miniIso(),float(0.99)), std::min(lep1_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep1_ptrel_v1,float(29.9)) );
 		}
 	      }
-	      if (ss.lep2_motherID()<=0 && /*ss.lep2_iso()>0.1 &&*/ fabs(ss.lep2_ip3d()/ss.lep2_ip3d_err())<4. && ss.lep1_motherID()==1) {
-		if (abs(ss.lep2_id())==11) {
-		  pTrelvsIso_histo_el->Fill( std::min(ss.lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_el->Fill( std::min(ss.lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+	      if (ss::lep2_motherID()<=0 && /*ss::lep2_iso()>0.1 &&*/ fabs(ss::lep2_ip3d()/ss::lep2_ip3d_err())<4. && ss::lep1_motherID()==1) {
+		if (abs(ss::lep2_id())==11) {
+		  pTrelvsIso_histo_el->Fill( std::min(ss::lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_el->Fill( std::min(ss::lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_el")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
 		} else {
-		  pTrelvsIso_histo_mu->Fill( std::min(ss.lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
-		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss.lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsIso_histo_mu->Fill( std::min(ss::lep2_iso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
+		  pTrelvsMiniIso_histo_mu->Fill( std::min(ss::lep2_miniIso(),float(0.99)), std::min(lep2_ptrel_v1,float(29.9)) );
 		  hists[getHist("pTrel_histo_mu")]->Fill(std::min(lep2_ptrel_v1,float(29.9)) );
 		}
 	      }
 	    }
 	  }
 
-  if (ss.hyp_class() == 3){
+
+  if (ss::hyp_class() == 3 && lep1_pT >= 20 && lep2_pT >= 20){
 
     counter++;
 
-    if (ss.lep1_passes_id()==0) continue;
-    if (ss.lep2_passes_id()==0) continue;
-    
```

Comment:
why are these removed? we want the observed, so both leptons must pass the selection (ok, maybe it is redundant)